source/common/upstream/outlier_detection_impl.cc

```diff
@@ -760,6 +756,22 @@ void DetectorImpl::onIntervalTimer() {
   processSuccessRateEjections(DetectorHostMonitor::SuccessRateMonitorType::ExternalOrigin);
   processSuccessRateEjections(DetectorHostMonitor::SuccessRateMonitorType::LocalOrigin);
 
+  // Decrement time backoff for all hosts which have not been ejected.
+  for (auto host : host_monitors_) {
+    if (!host.first->healthFlagGet(Host::HealthFlag::FAILED_OUTLIER_CHECK)) {
+      auto& monitor = host.second;
+      // Node is healthy and was not ejected since the last check.
+      if (monitor->lastUnejectionTime().has_value() &&
+          ((now - monitor->lastUnejectionTime().value()) >=
+           std::chrono::milliseconds(
+               runtime_.snapshot().getInteger(IntervalMsRuntime, config_.intervalMs())))) {
+        if (monitor->ejectTimeBackoff() != 0) {
+          monitor->ejectTimeBackoff()--;
+        }
+      }
+    }
+  }
```

Comment:
This is LGTM overall. But should we decrement the backoff directly if the host is healthy?

From the docs,
>  If the host is healthy during that check, the ejection time multiplier is decremented.

Reply:
That line of code checks is the node is healthy:

`
    if (!host.first->healthFlagGet(Host::HealthFlag::FAILED_OUTLIER_CHECK)) {
`

In outlier detection terms, HEALTHY means that it has not been ejected by one of outlier detection monitors yet. The logic here is: if the node was not ejected during last period of interval and is healthy now, we can start decrementing backoff time.

Reply:
I think you mis-understood it. But never mind. After a think again, I think you are right.

This is a reasonable change. There is no actual evidence that a host is healthy when a host is not be ejected. Because the host maybe just leave jail and haven't prove itself by processing enough traffic.

Reply:
cc @mattklein123 wdyt? I think you also take part in discussion in related issues.

Reply:
/wait-any

Reply:
> This is a reasonable change. There is no actual evidence that a host is healthy when a host is not be ejected. Because the host maybe just leave jail and haven't prove itself by processing enough traffic.

That is correct. Opposite to active HC, outlier is passive. Unless there is enough traffic to determine the node's state, the logic assumes that the node is healthy.

Thanks for reviewing!

Reply:
Yes this makes sense.